File: src/test/java/com/amaze/filemanager/test/FileAndFolderManipulationTests.java
package com.amaze.filemanager.test;

import com.amaze.filemanager.test.Utilities.TestDataSource;
import com.amaze.filemanager.test.Utilities.Utils;


import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.slf4j.helpers.Util;


import io.appium.java_client.TouchAction;

import static junit.framework.Assert.fail;

public class FileAndFolderManipulationTests extends BaseTestClass {

    private String fileName = TestDataSource.textFileName;
    private String folderName1 = TestDataSource.folderNames[0];
    private String folderName2= TestDataSource.folderNames[1];

    @Test
    public void testCreateNewFileAndDeleteIt(){
        //create the file
        Utils.createFileWithName(driver, fileName);

        //assert it exists
        if(driver.findElements(By.name(fileName)).isEmpty()){
            fail("File is not visible");
        }

        //delete the file
        WebElement folder = driver.findElementByName(fileName);
        TouchAction longPress = new TouchAction(driver);
        longPress.longPress(folder, 2000).release().perform();
        driver.findElementById("com.amaze.filemanager:id/delete").click();
        driver.findElementById("com.amaze.filemanager:id/buttonDefaultPositive").click();

        //assert the file is deleted
        if(!driver.findElements(By.name(fileName)).isEmpty()){
            fail("File is still visible, should be deleted");
        }

    }

    @Test
    public void testCreateNewFolderAndDeleteIt(){

        //create the folder
        Utils.createFolderWithName(driver, folderName1);

        //assert it exists
        if(driver.findElements(By.name(folderName1)).isEmpty()){
            fail("Folder is not visible");
        }

        //remove the folder
        WebElement folder = driver.findElementByName(folderName1);
        TouchAction longPress = new TouchAction(driver);
        longPress.longPress(folder, 2000).release().perform();
        driver.findElementById("com.amaze.filemanager:id/delete").click();
        driver.findElementById("com.amaze.filemanager:id/buttonDefaultPositive").click();

        //assert the folder is deleted
        if(!driver.findElements(By.name(folderName1)).isEmpty()){
            fail("Folder is still visible, should be deleted");
        }
    }

    @Test
    public void testCopyFileToAnotherFolder(){
        Utils.createFolderWithName(driver, folderName1);
        Utils.createFolderWithName(driver, folderName2);

        //add file to folder 1
        driver.findElementByName(folderName1).click();
        Utils.createFileWithName(driver, fileName);

        //copy the file to the other folder
        driver.findElementById("com.amaze.filemanager:id/properties").click();
        driver.findElementByName("Copy").click();

        driver.navigate().back();

        //have to swipe down to make the top part of toolbar visible
        Utils.swipeDownInPathBar(driver);

        //paste the file to another folder
        driver.findElementByName(folderName2).click();
        driver.findElementById("com.amaze.filemanager:id/paste").click();

        //wait for the slow copy action to perform
        WebDriverWait driverWait = new WebDriverWait(driver, 10);
        driverWait.until(ExpectedConditions.visibilityOfElementLocated(
                By.name(fileName)
        ));

        //assert the file is is visible
        if(driver.findElements(By.name(fileName)).isEmpty()){
            fail("File should be pasted in, is not visible on the screen");
        }
    }

}

File: src/tauTests/tests/FileAndFolderManipulationTests.py
# -*- coding: utf-8 -*-
from Utilities import TestDataSource, Utils

class FileAndFolderManipulationTests(UITestCase):
    
    def setUp(self):
        global folderName1
        global folderName2
        global testingFolderName
        global amazeTestingFolder
        global fileName
        
        testingFolderName = TestDataSource.generalTestFolderName
        folderName1 = TestDataSource.folderName1
        folderName2 = TestDataSource.folderName2
        amazeTestingFolder = TestDataSource.amazeTestFolderName
        fileName = TestDataSource.textFileName
        
        reactor.add("Permission Reactor", self.allowPermissions, text="Allow")
        launch.activity('com.amaze.filemanager', '.activities.MainActivity', verify=False)
        
        Utils.navigateToTestFolder()
        #Utils.createFolderWithName(amazeTestingFolder)
        tap.instanceOf('android.widget.ImageView', index=2)
        tap.resourceId("com.amaze.filemanager:id/menu_item")
        input.text(amazeTestingFolder)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        tap.text(amazeTestingFolder)
        
    def tearDown(self):
        packages.clearData('com.amaze.filemanager')
        #remove the testing folder
        shell('rm -r /storage/emulated/0/Testing/%s' % amazeTestingFolder)
        
    def allowPermissions(self):
        tap.resourceId("com.android.packageinstaller:id/permission_allow_button")
 
        
    @testCaseInfo('<Add a new folder>', deviceCount=1)
    def testCreateNewFolderAndDeleteIt(self):
      
        #create the folder
        #Utils.createFolderWithName(folderName1)
        #for some reason the way it is done in utils does not work!
        tap.instanceOf('android.widget.ImageView', index=2)
        tap.resourceId("com.amaze.filemanager:id/menu_item")
        input.text(folderName1)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
        #assert it is visible
        verify.text(folderName1)
        
        #delete the file
        tap.long.text(folderName1)
        tap.resourceId('com.amaze.filemanager:id/delete')
        tap.resourceId('com.amaze.filemanager:id/buttonDefaultPositive')
        
        #assert it is gone
        verify.no.text(folderName1)
        
    @testCaseInfo('<Add a new file>', deviceCount=1)
    def testCreateNewFileAndDeleteIt(self):
        
        #create folder, doesnt work if helper method is used
        #Utils.createFileWithName(fileName)
        tap.instanceOf('android.widget.ImageView', index=2)
        tap.resourceId("com.amaze.filemanager:id/menu_item1")
        input.text(fileName)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
        #assert file is visible
        #assert exists.text(fileName), \
        #"file " + fileName + " should be visible"
        verify.text(fileName)
        
        #delete the file
        tap.long.text(fileName)
        tap.resourceId('com.amaze.filemanager:id/delete')
        tap.resourceId('com.amaze.filemanager:id/buttonDefaultPositive')
        
        #assert it is gone
        #assert exists.no.text(fileName), \
        #"file " + fileName + " should not be visible"
        verify.no.text(fileName)
        
        
    @testCaseInfo('<Copy a file to another folder>', deviceCount=1)
    def testCopyFileToAnotherFolder(self):
        #create folders, doesnt work if helper methods are used
        #self.createFolderWithName(folderName1)
        tap.instanceOf('android.widget.ImageView', index=2)
        tap.resourceId("com.amaze.filemanager:id/menu_item")
        input.text(folderName1)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
        #self.createFolderWithName(folderName2)
        tap.instanceOf('android.widget.ImageView', index=2)
        tap.resourceId("com.amaze.filemanager:id/menu_item")
        input.text(folderName2)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
        #add file to the folder
        tap.text(folderName1)
        #self.createFileWithName(fileName)
        tap.instanceOf('android.widget.ImageView', index=2)
        tap.resourceId("com.amaze.filemanager:id/menu_item1")
        input.text(fileName)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
        #copy the file
        tap.resourceId("com.amaze.filemanager:id/properties")
        tap.text("Copy")
        
        input.key.back()
        Utils.swipeDownInPathBar()
        
        #paste the file to another folder
        tap.text(folderName2)
        tap.resourceId("com.amaze.filemanager:id/paste")
        
        #assert file is visible
        #assert exists.text(fileName, wait=10000), \
        #"file " + fileName + " should be visible"
        verify.text(fileName, wait=10000)
        
    
    def createFolderWithName(self, folderName):
        tap.instanceOf('android.widget.ImageView')
        tap.resourceId("com.amaze.filemanager:id/menu_item")
        input.text(folderName)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
    def createFileWithName(self, name):
        tap.instanceOf('android.widget.ImageView')
        tap.resourceId("com.amaze.filemanager:id/menu_item1")
        input.text(name)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
    